src/NUnitFramework/framework/Constraints/Comparers/IChainComparer.cs

```diff
@@ -36,10 +36,11 @@ internal interface IChainComparer
         /// <param name="x">The first object to compare.</param>
         /// <param name="y">The second object to compare.</param>
         /// <param name="tolerance">The tolerance to use when comparing the objects.</param>
+        /// <param name="topLevelComparison">Flag indicating whether or not this is the top level comparison.</param>
         /// <returns>
         ///     <c>null</c> if the objects cannot be compared using the method.
         ///     Otherwise the result of the comparison is returned.
         /// </returns>
-        bool? Equal(object x, object y, ref Tolerance tolerance);
+        bool? Equal(object x, object y, ref Tolerance tolerance, bool topLevelComparison = true);
```

Comment:
The only reason to use default values is for the convenience of the caller. Is `IChainCaller` directly called a lot? I have a small preference for making things explicit at the call site. We don't want to make it easy for future developers to miss the significance of the decision going into the parameter.

Reply:
I geneally feel the same way but I have to disagree in this case. The external caller, which actually makes the top level call, should not have to say "This is a top level call" and should not be able to say "This is not a top level call."

 Admittedly, exposing the argument this way is a bit of a compromise. I suggested it because it allows keeping the information about whether it's a top level call on the stack.